internal/impl/aws/input_sqs.go

```diff
@@ -283,47 +318,44 @@ func (a *awsSQSReader) ackLoop(wg *sync.WaitGroup, inFlightTracker *sqsInFlightT
 			}
 		} else {
 			if err := a.resetMessages(closeNowCtx, handles...); err != nil {
-				a.log.Errorf("Failed to reset the visibility timeout of messages: %v", err)
+				// Downgrade this to Info level - it's not really an error, it's just going to take longer
+				// to reset the visibility so the messages might be delayed is all. It's possible for delays
+				// if this succeeds anyways as it might be racing with the refresh loop. Fixing that
+				// would mean moving nacks to the refresh loop, but I don't think this will be a big deal in
+				// practice.
+				a.log.Infof("Failed to reset the visibility timeout of messages: %v", err)
 			}
 		}
 	}
 
-	refreshCurrentHandles := func() {
-		currentHandles, timeoutSeconds := inFlightTracker.PullToRefresh()
-		if len(currentHandles) == 0 {
-			return
-		}
-		if err := a.updateVisibilityMessages(closeNowCtx, timeoutSeconds, currentHandles...); err != nil {
-			a.log.Debugf("Failed to update messages visibility timeout: %v", err)
-		}
-	}
-
 	flushTimer := time.NewTicker(time.Second)
 	defer flushTimer.Stop()
 
-	// Both maps are of the message ID to the receipt handle
-	pendingAcks := map[string]string{}
-	pendingNacks := map[string]string{}
+	pendingAcks := []*sqsMessageHandle{}
+	pendingNacks := []*sqsMessageHandle{}
```

Comment:
Since these are lists of pointers now, do `deleteMessages()` and `resetMessages()` always get non-nil message handles? Might be worth adding some defensive nil checks in those methods.

Reply:
never nil 😆

Reply:
Hahah, yeah :)) I figured ❤️